aten/src/ATen/CPUGenerator.h

```diff
@@ -3,10 +3,11 @@
 #include <ATen/core/Generator.h>
 #include <ATen/core/MT19937RNGEngine.h>
 #include <c10/util/Optional.h>
+#include <c10/core/GeneratorImpl.h>
 
 namespace at {
 
-struct CAFFE2_API CPUGenerator : public Generator {
+struct CAFFE2_API CPUGenerator : public c10::GeneratorImpl {
```

Comment:
don't have to do it in this PR, but CPUGenerator and friends should probably also get the Impl suffix too

Reply:
#34984